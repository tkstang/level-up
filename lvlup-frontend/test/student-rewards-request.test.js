import StudentRewardRequest from '../src/components/Student/requests/request-main';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('StudentRewardRequest should render', () => {
  const store = mockStore({ lvlupApp });
  const req = shallow(
    <StudentRewardRequest reward={{ point_cost: 75 }} pts={{ currentTotal: 275 }} requestStatus={{ fulfilled: false }} store={store} />,
  );
  expect(shallowToJson(req)).toMatchSnapshot();
});
