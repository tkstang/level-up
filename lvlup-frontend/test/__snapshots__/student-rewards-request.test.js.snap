exports[`test StudentRewardRequest should render 1`] = `
<div>
  <Connect(ReduxForm) />
</div>
`;
